# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientInfo.h
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 dom/clients/manager/ClientInfo.h
--- a/dom/clients/manager/ClientInfo.h	Tue May 21 22:51:54 2019 +0000
+++ b/dom/clients/manager/ClientInfo.h	Tue May 21 23:14:27 2019 +0000
@@ -8,11 +8,13 @@
 #define _mozilla_dom_ClientInfo_h
 
 #include "mozilla/dom/ClientBinding.h"
+#include "mozilla/Maybe.h"
 #include "mozilla/UniquePtr.h"
 
 namespace mozilla {
 
 namespace ipc {
+class CSPInfo;
 class PrincipalInfo;
 }  // namespace ipc
 
@@ -87,6 +89,12 @@
   // Get a main-thread nsIPrincipal for the client.  This may return nullptr
   // if the PrincipalInfo() fails to deserialize for some reason.
   nsCOMPtr<nsIPrincipal> GetPrincipal() const;
+
+  const Maybe<mozilla::ipc::CSPInfo>& GetCspInfo() const;
+  void SetCspInfo(const mozilla::ipc::CSPInfo& aCSPInfo);
+
+  const Maybe<mozilla::ipc::CSPInfo>& GetPreloadCspInfo() const;
+  void SetPreloadCspInfo(const mozilla::ipc::CSPInfo& aPreloadCSPInfo);
 };
 
 }  // namespace dom